### DISCUSSION TITLE
Any suggestions on the best implementation of a tick event?

### DISCUSSION THREAD

**Discussion by JamesPatrickGill** [rockorager/libvaxis#142] [Q&A] [answered]
Hi I am new to this library (and zig infact) and was wondering if there's an example somewhere of a tick event being implemented? I am looking to have my app update on a set intervals with it being raised as the standard vaxis event on the the pollEvent function - any ideas?

**Comment by rockorager** [comment, marked as answer]
👋 Welcome!

You might check out the vxfw framework (src/vxfw/App.zig). This implements a .tick event, and you can look at Spinner.zig for how it is used to control animations

**Reply by JamesPatrickGill** [reply]
Thanks having a look now!
